enumerables.rb

```diff
@@ -0,0 +1,168 @@
+module Enumerable
+  # my_each
+  def my_eac
+    return enum_for(:my_each) unless block_given?
+
+    arr = self if self.class == Array
+    arr = to_a if self.class == Range || Hash
+    0.upto(arr.length - 1) do |index|
+      yield(arr[index])
+    end
+    arr
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    arr = self if self.class == Array
+    arr = to_a if self.class == Range || Hash
+
+    0.upto(arr.size - 1) do |index|
+      yield(arr[index], index)
+    end
+    arr
+  end
+
+  # my_select
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    new_a = []
+    my_each do |element|
+      new_a << element if yield(element)
+    end
+    new_a
+  end
+
+  # My_all?
+  def my_all?(parameter = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false }
+      return true
+    elsif parameter.nil?
+      my_each { |n| return false if n.nil? || n == false }
+    elsif !parameter.nil? && (parameter.is_a? Class)
+      my_each { |n| return false unless [n.class, n.class.superclass].include?(parameter) }
+    elsif !parameter.nil? && patameter.class == Regexp
+      my_each { |n| return false unless parameter.match(n) }
+    else
+      my_each { |n| return false if n != parameter }
+    end
+    true
+  end
+
+  # my_any?
+  def my_any?(parameter = nil)
+    if block_given?
+      my_each { |item| return true if yield(item) }
+      false
+    elsif parameter.nil?
+      my_each { |n| return true if n }
+    elsif !parameter.nil? && (parameter.is_a? Class)
+      my_each { |n| return true if [n.class, n.class.superclass].include?(parameter) }
+    elsif !parameter.nil? && parameter.class == Regexp
+      my_each { |n| return true if parameter.match(n) }
+    else
+      my_each { |n| return true if n == parameter }
+    end
+    false
+  end
+
+  # none?
+  def my_none?
+    return enum_for(:my_none) unless block_given?
+
+    my_each do |element|
+      return false if yield(element)
+    end
+  end
```

Comment:
- my_none? returns true if the block never returns true for all elements
- my_none? when no block or argument is given returns true only if none of the collection members is true
- my_none? when no block or argument is given returns false only if one of the collection members is true
- my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
- my_none? when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric
- my_none? when a class is passed as an argument returns false if any of the collection is a member of such class
- my_none? when a Regex is passed as an argument returns true if none of the collection matches the Regex
- my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex
- my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
- my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern